sql/core/src/main/scala/org/apache/spark/sql/execution/LocalTableScan.scala

```diff
@@ -32,9 +32,20 @@ case class LocalTableScan(output: Seq[Attribute], rows: Seq[Row]) extends LeafNo
 
   override def execute(): RDD[Row] = rdd
 
-  override def executeCollect(): Array[Row] =
-    rows.map(ScalaReflection.convertRowToScala(_, schema)).toArray
 
-  override def executeTake(limit: Int): Array[Row] =
-    rows.map(ScalaReflection.convertRowToScala(_, schema)).take(limit).toArray
+  override def executeCollect(): Array[Row] = {
+    val converters = schema.fields.map {
```

Comment:
Is there a reason this pattern cannot be

```scala
val converter = CatalystTypeConverters.createToScalaConverter(schema)
rows.map(converter).toArray
```
?

It looks like this pretty efficiently handles this situation in the same way that you've extracted it here (even with calling convertRowWithConverters).